lib/src/block_parser.dart

```diff
@@ -144,6 +143,20 @@ abstract class BlockSyntax {
     if (parser.isDone) return true;
     return parser.blockSyntaxes.any((s) => s.canParse(parser) && s.canEndBlock);
   }
+
+  /// Generate a valid HTML anchor from the inner text of [element].
+  static String generateAnchorHash(Element element) =>
+      _concatenatedText(element)
+          .toLowerCase()
+          .replaceAll(new RegExp(r'^[^a-z]+'), '')
```

Comment:
This line kills all leading non-alphabetical characters. The next line kills all non alphanumeric+`[ _-]` characters, throughout. I don't think they're combinable.